firmware/hw_layer/debounce.cpp

```diff
@@ -6,16 +6,60 @@
  * @author David Holdeman, (c) 2020
  */
 #include "debounce.h"
+#include "pin_repository.h"
+#include "engine_configuration.h"
+#include "hardware.h"
 
-void ButtonDebounce::init (int t, brain_pin_e p, pin_input_mode_e m) {
+ButtonDebounce* ButtonDebounce::s_firstDebounce = nullptr;
+
+void ButtonDebounce::init (int t, brain_pin_e *p, pin_input_mode_e *m) {
```

Comment:
nit: name these `pin` and `mode`

Reply:
sure. Golang experience kicking in on me.